test/unit/traversing.js

```diff
@@ -430,6 +430,23 @@ QUnit.test( "not(jQuery)", function( assert ) {
 	assert.deepEqual( jQuery( "p" ).not( jQuery( "#ap, #sndp, .result" ) ).get(), q( "firstp", "en", "sap", "first" ), "not(jQuery)" );
 } );
 
+QUnit.test( "contents().not(':not(:not(*)')", function( assert ) {
+	assert.expect( 1 );
+
+	var elem = jQuery( "<div>foo<b>bar</b></div>" );
+
+	assert.deepEqual( elem.contents().not( ":not(:not(*))" ).get(), [ ], "contents().not(':not(:not(*))')" );
+} );
+
+QUnit.test( "contents().not(':not(:not(:not(*))')", function( assert ) {
```

Comment:
Can you combine these tests together into something like `"not(Selector) excludes non-element nodes (gh-2808)"`, and use simpler selectors? E.g.,
```js
var mixedContents = jQuery( "#nonnodes" ).contents(),
	childElements = q( "nonnodesElement" );
assert.deepEqual( mixedContents.not( "*" ).get(), [], "not *" );
assert.deepEqual( mixedContents.not( "[id=a],[id=b]" ).get(), childElements, "not [id=a],[id=b]" );
assert.deepEqual( mixedContents.not( "[id=a],*,[id=b]" ).get(), [], "not [id=a],*,[id=b]" );
```
(https://github.com/vanderlee/jquery/blob/patch-1/test/index.html#L210)